pil-analyzer/src/statement_processor.rs

```diff
@@ -512,19 +520,27 @@ where
         })]
     }
 
-    fn expression_processor(&self) -> ExpressionProcessor<D> {
-        ExpressionProcessor::new(self.driver)
+    fn expression_processor<'b>(
+        &'b self,
+        type_vars: &'b HashSet<&'b String>,
+    ) -> ExpressionProcessor<'b, D> {
+        ExpressionProcessor::new(self.driver, type_vars)
     }
 
-    fn process_expression(&self, expr: parsed::Expression) -> Expression {
-        self.expression_processor().process_expression(expr)
+    fn process_expression(
+        &self,
+        expr: parsed::Expression,
+        type_vars: &HashSet<&String>,
+    ) -> Expression {
+        self.expression_processor(type_vars)
+            .process_expression(expr)
```

Comment:
nitpick: should there be wrapping methods for the other `.expression_processor().process_foo` calls?

Reply:
Do you have an example? There are other wrapping methods as well, for example process_selected_expression.

Reply:
Oh, I just realized I don't use them.

Reply:
Ah no, it's other cases. Yeah, maybe I should also remove the helpers, it's kind of cluttery.